src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/SwitchHandler.java

```diff
@@ -81,10 +81,20 @@ private void checkSwitchExpr() {
     protected IndentLevel getIndentImpl() {
         IndentLevel indentLevel = super.getIndentImpl();
         // if switch is starting the line
-        if (isOnStartOfLine(getMainAst())
-                && TokenUtil.isOfType(getMainAst().getParent().getParent(), TokenTypes.ASSIGN)) {
-            indentLevel = new IndentLevel(indentLevel,
-                    getIndentCheck().getLineWrappingIndentation());
+        if (isOnStartOfLine(getMainAst())) {
+            final DetailAST parent = getMainAst().getParent();
+            final DetailAST grandParent = parent.getParent();
+            
+            // Check for assignment context (from PR #16418)
+            if (TokenUtil.isOfType(grandParent, TokenTypes.ASSIGN)) {
+                indentLevel = new IndentLevel(indentLevel,
+                        getIndentCheck().getLineWrappingIndentation());
+            }
+            // Check for lambda context
+            else if (TokenUtil.isOfType(grandParent, TokenTypes.LAMBDA)) {
+                indentLevel = new IndentLevel(indentLevel,
+                        getIndentCheck().getLineWrappingIndentation());
+            }
```

Comment:
```suggestion
            // Check for assignment and lambda context (from PR #16418)
            if (TokenUtil.isOfType(grandParent, TokenTypes.ASSIGN) || TokenUtil.isOfType(grandParent, TokenTypes.LAMBDA)) {
                indentLevel = new IndentLevel(indentLevel,
                        getIndentCheck().getLineWrappingIndentation());
            }
```
avoid dry if possible